ground/gcs/src/plugins/config/configgadgetwidget.cpp

```diff
@@ -146,8 +183,24 @@ ConfigGadgetWidget::ConfigGadgetWidget(QWidget *parent) : QWidget(parent)
         onAutopilotConnect();    
     }
 
-    help = 0;
     connect(ftw,SIGNAL(currentAboutToShow(int,bool*)),this,SLOT(tabAboutToChange(int,bool*)));//,Qt::BlockingQueuedConnection);
+    return;
+    }
+
+    chunk++;
+
+    QTimer::singleShot(0, this, SLOT(deferredLoader()));
+}
+
+void ConfigGadgetWidget::paintEvent(QPaintEvent *event)
+{
+    (void) event;
+
+    if (chunk < 12) {
+	// jumpstart loading.
+        deferredLoader();
```

Comment:
Yes.  That way we do 2 steps per chunk in the event loop (or more, if the deferred loading has already started).  This effectively means "hurry up."